### PR TITLE
Use subqueries to retrieve collection-level data

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/models/concerns/arclight/solr_document.rb [hunk 1]
```diff
@@ -34,12 +28,8 @@ def parent_levels
...
     end
 
-    def parent_document
-      self.class.new fetch('parent').fetch('docs', []).first
-    end
-
     def eadid
-      fetch('ead_ssi', nil)&.strip
+      first('ead_ssi')&.strip || collection&.first('ead_ssi')&.strip
     end
```
Comment (jcoyne) [inline app/models/concerns/arclight/solr_document.rb:33]: can you add a comment about why you fall back to looking at the collection for ead_id?

File: app/models/concerns/arclight/solr_document.rb [hunk 2]
```diff
@@ -51,19 +41,25 @@ def unitid
...
 
     def repository_and_unitid
       [repository, unitid].compact.join(': ')
     end
 
+    def collection
+      return self if collection?
+
+      @collection ||= self.class.new(self['collection']&.dig('docs', 0), @response)
```
Comment (jcoyne) [inline app/models/concerns/arclight/solr_document.rb:54]: can you add a comment about what this is doing?

File: app/models/concerns/arclight/solr_document.rb [hunk 3]
```diff
@@ -107,7 +103,7 @@ def level
     end
 
     def collection?
-      level.parameterize == 'collection'
+      level&.parameterize == 'collection'
```
Comment (jcoyne) [inline app/models/concerns/arclight/solr_document.rb:106]: why wouldn't this have a level?

File: app/models/concerns/arclight/solr_document.rb [hunk 4]
```diff
@@ -51,19 +41,25 @@ def unitid
...
 
+    def collection
+      return self if collection?
+
+      @collection ||= self.class.new(self['collection']&.dig('docs', 0), @response)
+    end
+
     def collection_name
-      first('collection_ssm')
+      collection&.normalized_title
```
Comment (jcoyne) [inline app/models/concerns/arclight/solr_document.rb:58]: is there a reason it wouldn't have a collection?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/models/concerns/arclight/solr_document.rb (hunk 3 above)
Comment (cbeer) [reply app/models/concerns/arclight/solr_document.rb:106]: It's probably bad test data, but it's not impossible 🤷‍♂️

File: app/models/concerns/arclight/solr_document.rb (hunk 4 above)
Comment (cbeer) [reply app/models/concerns/arclight/solr_document.rb:58]: Most likely, bad fixture data. There's some indication in the indexing code that the top-level EAD might be something other than a collection, though, so it didn't feel too inappropriate to me 🤷‍♂️

**Review by jcoyne — APPROVED** [review]

**Pull request merged**